sympy/physics/mechanics/functions.py

```diff
@@ -532,7 +539,7 @@ def _recurser(expr, sub_dict):
 
 def _fraction_decomp(expr):
     """Return num, den such that expr = num/den"""
-    if not isinstance(expr, Mul):
+    if not isinstance(expr, MulClass):
```

Comment:
In SymPy, `Add(x, x)` works and returns a `Mul`. Also `x + y` is an instance of `Add`.

But for SymEngine, we can't have both. Hence `Add` is the class and `add` will sum all the terms given.